### PR TITLE
Fix Tests/versions and unicode 

### PR DESCRIPTION
**Pull request by srossross**

### CODE REVIEW

File: binstar_build_client/worker/tests/test_build_log.py [hunk 1]
```diff
@@ -17,10 +19,27 @@ def tearDown(self):
         os.unlink(self.filepath)
 
     def test_write(self):
-        with BuildLog(BSClient(), "un", "queue", "worker_id", 123, filename=self.filepath) as log:
```
Comment (rafaelhbarros) [inline binstar_build_client/worker/tests/test_build_log.py]: why removing the `with` from here?

Comment (rafaelhbarros) [inline binstar_build_client/worker/tests/test_build_log.py]: nevermind

File: versioneer.py [hunk 2]
```diff
@@ -975,7 +975,7 @@ def git_versions_from_keywords(keywords, tag_prefix, verbose):
         # "stabilization", as well as "HEAD" and "master".
         tags = set([r for r in refs if re.search(r'\d', r)])
         if verbose:
-            print("discarding '%s', no digits" % ",".join(refs-tags))
+            print("discarding '%s', no digits" % ",".join(refs - tags))
```
Comment (rafaelhbarros) [inline versioneer.py]: `.format`

Comment (srossross) [inline versioneer.py]: This versioneer code that my editor formatted + a conda integration point. I am not going to fix this.

File: binstar_build_client/worker/utils/timeout.py [hunk 3]
```diff
@@ -75,8 +75,7 @@ def iotimer():
 
         while line:
             iotimer.tick()
-            if isinstance(line, str):
-                line = line.encode('utf8', errors='ignore')
```
Comment (rafaelhbarros) [inline binstar_build_client/worker/utils/timeout.py]: causing:
```
File "/Users/rdebarros/work/anaconda-build/binstar_build_client/worker/utils/timeout.py", line 79, in read_with_timeout
    output.write(line)
TypeError: a bytes-like object is required, not 'str'

Comment (srossross) [inline binstar_build_client/worker/utils/timeout.py]: Is fixed now.

**Comment by rafaelhbarros** [comment]
LGTM :+1:

**Pull request merged**